code/game/machinery/doors/airlock.dm

```diff
@@ -1200,6 +1198,7 @@
 
 	if(!density)
 		return TRUE
+	ui_update()
```

Comment:
Does the UI need to update afeter the sleeps() too?

Reply:
I added further updates to both `open` and `close`, please check if those look fine now.